### PR TITLE
Support aggregation via the horizontal-multivec selectRows option

### PR DESCRIPTION
**Pull request by keller-mark**
## Description

> What was changed in this pull request?

This adds support for passing a 2d / mixed 1d-2d array to the `selectRows` track option of the horizontal-multivec track.

It also adds the options 
- `selectRowsAggregationMode` which takes a string "mean", "sum", "variation", "deviation" to determine which aggregation function to use. Right now I am using the d3-array implementations of these four functions.
- `selectRowsAggregationWithRelativeHeight`, a boolean value which determines whether the visual height of an aggregation group is relative to the group size. 

For example, if `selectRows` is `[ [1, 2, 3], [4], 5 ]` then the canvas will always have 3 total rows.
But the first row will be 3x the size of the other two if `selectRowsAggregationWithRelativeHeight` is `true`. Otherwise, all 3 rows will have uniform height.

> Why is it necessary?

This will allow aggregation to be performed for multiple rows that can be grouped based on some external metadata value, for example cell type or tissue type in the context of the Cistrome ChIP-seq samples.

Fixes #___

## Checklist

- [ ] Set proper GitHub labels (e.g. v1.6+, ignore if you don't know)
- [x] Unit tests added or updated
- [x] Documentation added or updated
- [ ] Example(s) added or updated
- [ ] Update schema.json if there are changes to the viewconf JSON structure format
- [x] Screenshot for visual changes (e.g. new tracks or UI changes)
- [x] Updated CHANGELOG.md

### CODE REVIEW

File: app/scripts/utils/get-aggregation-function.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      break;
+    case 'variance':
+      aggFunc = variance;
+      break;
+    case 'deviation':
+      aggFunc = deviation;
+      break;
+    default:
+      console.warn(
+        'Encountered an unsupported selectedRowsAggregationMode option.'
```
Comment (pkerpedjiev) [inline app/scripts/utils/get-aggregation-function.js]: Crash out or default to mean?

**Comment by keller-mark** [comment]
Screenshots:

Example 1: sum with relative heights
```
{ // horizontal-multivec track options
...,
"selectRows": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]],
"selectRowsAggregationMode": "sum",
"selectRowsAggregationWithRelativeHeight": true
}
```

Screen Shot 2020-03-12 at 11 21 49 AM

Example 2: sum with uniform heights
```
{ // horizontal-multivec track options
...,
"selectRows": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]],
"selectRowsAggregationMode": "sum",
"selectRowsAggregationWithRelativeHeight": false
}
```

Screen Shot 2020-03-12 at 11 25 30 AM

File: app/scripts/HorizontalMultivecTrack.js [hunk 2]
```diff
@@ -13,12 +16,28 @@ export default class HorizontalMultivecTrack extends HeatmapTiledPixiTrack {
     this.continuousScaling = false;
   }
 
+  rerender(options, force) {
+    super.rerender(options, force);
+
+    if (this.options.selectRows) {
+      // The weights for selectRows groups must be computed
```
[the remark this answers is not in the document]
Comment (keller-mark) [reply app/scripts/HorizontalMultivecTrack.js]: These just help to figure out the row group over which the mouse is positioned in order to compute the value for the tooltip, even when row heights are not uniform.

For example, if the following is the selectRows array `[[1,2,3],4,5]` and the relativeHeight option is `true`, then the first row will be 3x the size of the others. So we cannot just do `Math.floor((mouseY/trackHeight)*numRows)` to figure out which item of the `selectRows` array that the mouseY corresponds to. Instead, if we have the array of precomputed cumulative weights, we can compute `mouseY / trackHeight` to get a normalized y position of the mouse and then check where in the cumulative weights array that the normalized y position value falls below the weight value and the next item will be too large.

[the remark this answers is not in the document]
Comment (pkerpedjiev) [reply app/scripts/HorizontalMultivecTrack.js]: 👍

**Pull request merged**